distributed/utils.py

```diff
@@ -1555,6 +1555,9 @@ def __setitem__(self, key, value):
             self.data.popitem(last=False)
         super().__setitem__(key, value)
 
+    def clear(self):
```

Comment:
This seems dangerous -- in `self.data` is where workers will store a MutabbleMapping of keys and values -- essentially all the data on a worker.  Perhaps `importlib.reload` might be a better/safer option here

Reply:
@quasiben thanks for the feedback!

in this particular case, the `LRU.clear` method is only called on `worker.cache_loads`, a LRU instance.

`cache_loads` is a **pass-thru cache** of mapping of pickled functions to the their loaded (unpickled) function objects. Besides the `clear` method, here is the only place it is set and accessed: https://github.com/dask/distributed/blob/master/distributed/worker.py#L3270-L3273

`cache_loads` seems to be a redundant state of functions only, and pickled function and args are passed in with each delayed call, but this cache aims to alleviate the burden of loading the pickled function every time by caching the already loaded function object. 

the problem, which this PR is meant to address, is that some of these function objects can be outdated after each file upload. 

Perhaps I am missing something that is potential harmful?

Reply:
@quasiben thanks for the feedback!

in this particular case, the LRU.clear method is only called on worker.cache_loads, a LRU instance.

cache_loads is a pass-thru cache of mapping of pickled functions to the their loaded (unpickled) function objects. Besides the clear method, here is the only place it is set and accessed: https://github.com/dask/distributed/blob/master/distributed/worker.py#L3270-L3273

cache_loads seems to be a redundant state of functions only, and pickled function and args are passed in with each delayed call, but this cache aims to alleviate the burden of loading the pickled function every time by caching the already loaded function object.

the problem, which this PR is meant to address, is that some of these function objects can be outdated after each file upload.

Perhaps I am missing something that is potential harmful?

Reply:
Nope, you're good. I misread the the self.data.clear line and thought this was happening in the worker, not the LRU cache.

Reply:
I recommend that we remove this method and just use `cache_loads.data.clear()` instead.  

Motivation: http://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection